llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -411,6 +417,21 @@ std::string saveModuleIR(Module &M, int I, StringRef Suff) {
   return OutFilename;
 }
 
+bool isImportedFunction(const Function &F) {
+  // Functions with definitions are not imported
+  if (!F.isDeclaration())
+    return false;
+
+  bool ReturnValue = true;
+  if (auto NameStr = itaniumDemangle(F.getName())) {
```

Comment:
May be auto &NameStr; we can avoid 'free' then?

Reply:
I don't think using an auto reference will automatically free a pointer.  As far as I can tell, all the uses of itaniumDemangle are followed by a free or the demangled string is kept in a structure in the LLVM source base.